### PR TITLE
Feat: View details displays in active requests

### PR DESCRIPTION
**Pull request by Tulun**
**Motivation**
User should see some summary data in view details

**Summary**

Modal opens to display summary data, including anc data. 

Also added some general refactoring for shared code.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [x]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: src/features/vote/useTableValues.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+interface TableValue {
+  ancillaryData: string;
+  identifier: string;
+  vote: string;
+  revealed: boolean;
+  ancHex: string;
+  timestamp: string;
+}
+
+export default function useTableValues(
```
Comment (daywiss) [inline src/features/vote/useTableValues.ts:19]: was this just pulled out of a component? if so what was changed?

**Review by daywiss — COMMENTED** [review]
code looks good, but ran into a bug:
- after connecting to wallet, app crashes when clicking on details for past request ( was connecting with mainnet)

image

- details should say what we are voting on, ie Price Request for identifier, or Governance Vote. is that possible? might require @Tommy1231232

File: src/features/vote/useTableValues.ts (hunk 1 above)
Comment (Tulun) [reply src/features/vote/useTableValues.ts:19]: Basically, the table values in both Commit and Reveal are super similar. So combined them into a hook for DRY.

File: src/features/vote/CommitPhase.tsx [hunk 2]
```diff
@@ -113,6 +122,7 @@ const ActiveRequestsForm: FC<Props> = ({
           let vc = votingContract;
           if (designatedVotingContract) vc = designatedVotingContract;
           if (vc) {
+            console.log("FD", fd, publicKey);
```
Comment (daywiss) [inline src/features/vote/CommitPhase.tsx:125]: log

File: src/features/vote/PastRequests.tsx [hunk 3]
```diff
@@ -70,6 +70,7 @@ const PastRequests: FC<Props> = ({
         Promise.all(pr).then((res) => {
           setPastRequests(!showAll ? res.slice(0, 10) : res);
         });
+        console.log("pr", pr);
```
Comment (daywiss) [inline src/features/vote/PastRequests.tsx:73]: log

File: src/features/vote/PastRequests.tsx [hunk 4]
```diff
@@ -70,6 +70,7 @@ const PastRequests: FC<Props> = ({
         Promise.all(pr).then((res) => {
           setPastRequests(!showAll ? res.slice(0, 10) : res);
         });
```
Comment (daywiss) [inline src/features/vote/PastRequests.tsx:72]: this needs a catch

**Review by daywiss — COMMENTED** [review]
looks good just a couple nits

File: src/features/vote/PastRequests.tsx (hunk 4 above)
Comment (Tulun) [reply src/features/vote/PastRequests.tsx]: Doesn't even need to be a promise >.< fixed.

File: src/features/vote/useTableValues.ts [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+      const tv = [] as TableValue[];
+      const postData = [] as PostRevealData[];
+
+      activeRequests.forEach((el) => {
+        const datum = {} as TableValue;
+        datum.ancillaryData = el.ancillaryData;
+        datum.identifier = el.identifier;
+        let vote = "-";
+        // I believe latest events are on bottom. requires testing.
+        const latestVotesFirst = [...encryptedVotes].reverse();
```
Comment (daywiss) [inline src/features/vote/useTableValues.ts:62]: this keeps moving back into the loop. dont think it needs to keep being re-created. if you are planning on refactoring this logic though dont worry about it in this PR

**Review by daywiss — APPROVED** [review]
LG thanks

**Pull request merged**